addons/currency_rate_live/models/res_company.py

```diff
@@ -0,0 +1,37 @@
+# -*- coding: utf-8 -*-
+import datetime
+from dateutil.relativedelta import relativedelta
+from openerp import api, fields, models
+
+
+class Company(models.Model):
+    _inherit = "res.company"
+
+    currency_interval_unit = fields.Selection([
+        ('daily', 'Daily'),
+        ('weekly', 'Weekly'),
+        ('monthly', 'Monthly')],
+        string='Interval Unit')
+    currency_provider_ids = fields.One2many('currency.rate.provider', 'company_id')
+    currency_next_execution_date = fields.Date(string="Next Execution Date")
+
+    @api.multi
+    def button_currency_update(self):
+        providers = self.env['currency.rate.provider'].search([('company_id', 'in', self.ids)])
+        for provider in providers:
+            provider.server_action_id.with_context(active_id=provider.id, active_model=provider._name).run()
+        for company in self:
+            if not company.currency_interval_unit:
+                continue
```

Comment:
It would be nice to avoid continue. 
```python
next_update = False
...
....
if next_update:
     company.currency_next_execution_date = datetime.datetime.now() + next_update